virttest/utils_disk.py

```diff
@@ -182,6 +183,585 @@ def clean_old_image(image):
+
+    :param session: session object to guest.
+    :param did: disk index which show in 'diskpart list disk'.
+    :param labeltype: label type for the disk.
+    """
+    if labeltype == "gpt":
+        mklabel_cmd = ' echo convert gpt >> disk '
+        mklabel_cmd = wrap_windows_cmd(mklabel_cmd)
+        session.cmd_output_safe(mklabel_cmd % did)
+
+
+def create_partition_table(session, did, ostype, labeltype="gpt"):
+    """
+    Create partition table on disk in linux or windows guest.
+
+    :param session: session object to guest.
+    :param did: disk kname or disk index
+    :param labeltype: label type for the disk.
+    :param ostype: linux or windows.
+    """
+    if ostype == "windows":
+        create_partition_table_windows(session, did, labeltype)
+    else:
+        create_partition_table_linux(session, did, labeltype)
+
+
+def create_partition_linux(session, did, start, end):
+    """
+    Create single partition on disk in linux guest.
+
+    :param session: session object to guest.
+    :param did: disk kname. e.g. sdb
+    :param start: partition beginning at start. e.g. 0
+    :param end: partition ending at end. e.g. size 200M
+    """
+    devname = '/dev/%s' % did
+    mkpart_cmd = "parted -s %s mkpart primary %s %s"
+    session.cmd_output_safe(mkpart_cmd % (devname, start, end))
+    session.cmd_output_safe("partprobe %s" % devname)
+
+
+def create_partition_windows(session, did, start, end, part_type="primary"):
+    """
+    Create single partition on disk in windows guest.
+
+    :param session: session object to guest.
+    :param did: disk index
+    :param start: partition beginning at start. e.g. 0
+    :param end: partition ending at end. e.g. size 200M
+    :param part_type: partition type. primary, extended, logical
+    """
+    size = end - start
+    mkpart_cmd = " echo create partition %s size=%s >> disk "
+    mkpart_cmd = wrap_windows_cmd(mkpart_cmd)
+    session.cmd(mkpart_cmd % (did, part_type, size))
+
+
+def create_partition(session, did, ostype, start, end):
+    """
+    Create single partition on disk in windows or linux guest.
+
+    :param session: session object to guest.
+    :param did: disk kname or disk index
+    :param ostype: linux or windows.
+    :param start: partition beginning at start. e.g. 0
+    :param end: partition ending at end. e.g. size 2G
+    """
+    end = utils_numeric.normalize_data_size(end, order_magnitude="M")
+    if ostype == "windows":
+        create_partition_windows(session, did, start, int(float(end)))
+    else:
+        create_partition_linux(session, did, start, end+"M")
+
+
+def create_partition_batch_linux(session, did, size, n_partitions):
+    """
+    Create multi-partitions on disk in linux guest.
+
+    :param session: session object to guest.
+    :param did: disk kname. e.g. sdb
+    :param size: partition size. e.g. 200M
+    :param n_partitions: the number of partitions on disk
+    """
+    partition_size = float(size[0:-1]) / n_partitions
```

Comment:
Better have `from __future__ import division`

Reply:
size is string, get from normalize_data_size()